### PR TITLE
[State Observability] Support --detail flag.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR adds --detail flag to the list APIs. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/state/state_head.py [hunk 1]
```diff
@@ -43,15 +44,20 @@ def __init__(self, dashboard_head):
 
     def _options_from_req(self, req: aiohttp.web.Request) -> ListApiOptions:
         """Obtain `ListApiOptions` from the aiohttp request."""
-        limit = int(req.query.get("limit"))
-        timeout = int(req.query.get("timeout"))
+        # limit = convert_string_to_type(req.query.get("limit"), int)
```
Comment (scv119) [inline dashboard/modules/state/state_head.py:47]: remove  comments?

File: python/ray/experimental/state/common.py [hunk 2]
```diff
@@ -72,16 +73,106 @@ class SummaryApiOptions:
...
+        for f in fields(cls):
+            if f.metadata.get("filterable", False):
+                filterable.add(f.name)
+        return filterable
 
-        NOTE: Currently, only bool, str, int, and float
-            types are supported for filtering.
-        TODO(sang): Filter on the source side instead for optimization.
+    @classmethod
+    def min_columns(cls) -> Set[str]:
```
Comment (scv119) [inline python/ray/experimental/state/common.py:140]: base_columns?

File: python/ray/experimental/state/util.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from typing import Union
+
+
+def convert_string_to_type(
+    val: Union[str, int, float, bool], convert_type: Union[int, float, bool]
```
Comment (scv119) [inline python/ray/experimental/state/util.py:5]: is the type annotation for convert_type correct?

**Review by scv119 — APPROVED** [review]

File: python/ray/experimental/state/util.py (hunk 3 above)
Comment (rkooo567) [reply python/ray/experimental/state/util.py:5]: I had to add Optional to the `val`. Good catch!

**Pull request merged**